sale_advance_payment/tests/test_sale_advance_payment.py

```diff
@@ -0,0 +1,247 @@
+# Copyright (C) 2021 ForgeFlow S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html)
+
+import json
+
+from odoo.tests import common
+
+
+class TestSaleAdvancePayment(common.SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        # Partners
+        cls.res_partner_1 = cls.env["res.partner"].create({"name": "Wood Corner"})
+        cls.res_partner_address_1 = cls.env["res.partner"].create(
+            {"name": "Willie Burke", "parent_id": cls.res_partner_1.id}
+        )
+        cls.res_partner_2 = cls.env["res.partner"].create({"name": "Partner 12"})
+
+        # Products
+        cls.product_1 = cls.env["product.product"].create(
+            {"name": "Desk Combination", "invoice_policy": "order"}
+        )
+        cls.product_2 = cls.env["product.product"].create(
+            {"name": "Conference Chair", "invoice_policy": "order"}
+        )
+        cls.product_3 = cls.env["product.product"].create(
+            {"name": "Repair Services", "type": "service", "invoice_policy": "order"}
+        )
+
+        cls.tax = cls.env["account.tax"].create(
+            {"name": "Tax 15", "type_tax_use": "sale", "amount": 20}
+        )
+
+        # Sale Order
+        cls.sale_order_1 = cls.env["sale.order"].create(
+            {"partner_id": cls.res_partner_1.id}
```

Comment:
```suggestion
            {"name": "/", partner_id": cls.res_partner_1.id}
```

I think CI fails because the module sale_order_revision works if you create a sales order as a user, but it does not if you create a sales order using write. If the name is "/" it will generate a new number for the sales order, so the constraint will not raise.

Can you try that?

Reply:
It still fails...it is weird because it is only failing in the OCB tests :/

Reply:
The same error?